enumerable.rb

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env ruby
+
+# Some methods will be added to this module
+module Enumerable # rubocop:disable Metrics/ModuleLength
+  # definition of the my_each method
+
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    for element in self do # rubocop:disable Style/For
+      yield(element)
+    end
+    self
+  end
+
+  # definition of the my_each_with_index method
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    index = 0
+    my_each do |element|
+      yield(element, index)
+      index += 1
+    end
+    self
+  end
+
+  # definition of the my_select method
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    new_array = []
+    my_each do |element|
+      new_array << element if yield(element)
+    end
+    new_array
+  end
+
+  # definition of the my_all? method
+
+  def my_all?(pattern = nil) # rubocop:disable CyclomaticComplexity, PerceivedComplexity
+    return true unless block_given? || !pattern.nil?
+
+    my_each do |element|
+      if block_given?
+        return false unless yield(element)
+      elsif pattern.is_a?(Class) || pattern.is_a?(Regexp)
+        return false unless pattern === element # rubocop:disable Style/CaseEquality
+      else
+        return false unless element == pattern
+      end
+    end
+    true
+  end
+
+  # definition of the my_none? method
+
+  def my_none?(pattern = nil) # rubocop:disable CyclomaticComplexity, PerceivedComplexity
```

Comment:
* [ ] `my_none?`  should identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
currently it is returning this on every test
![Screenshot from 2020-02-20 16-27-54](https://user-images.githubusercontent.com/29695661/74980177-0215c080-53fe-11ea-9b6e-ed7367a55120.png)
check line 61, ```return false unless block_given || !pattern.nil?```
after you have rectified the error go to the [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F) and run the test there to make sure they ```my_none``` is behaving the same as its clone ```none?```